lib/rspec/core/memoized_helpers.rb

```diff
@@ -153,6 +153,17 @@ def initialize
 
         def fetch_or_store(key)
           @memoized.fetch(key) do # only first access pays for synchronization
+            if @mutex.locked? && !@mutex.owned?
+              raise <<-EOS
+Possible deadlock at:
+  #{CallerFilter.first_non_rspec_line}
+
+This is normally caused when accessing an uninitialized `let` value in a thread
+different than the one calling `subject`. Either use `let!` or disable RSpec
```

Comment:
I'd reword this slightly to:

> This is normally caused when accessing an uninitialized `let` value in a different thread. Either use call the let before accessing it or disable RSpec threadsafety with `RSpec.configuration.threadsafe = false`.

Why? `subject` isn't used by everyone (and is just a special `let`) so making it more general should be a good thing, and as `let!` is just the equivalent of `let(:name) {}; before { name }` we have historically discouraged it's use in favour of more direct control.